src/compiler/interface.jl

```diff
@@ -139,7 +140,22 @@ end
 
 Base.show(io::IO, ps::Grads) = print(io, "Grads(...)")
 
-@forward Grads.grads Base.getindex, Base.haskey
+@forward Grads.grads  Base.haskey, Base.setindex!
```

Comment:
Is it correct to forward `haskey` here (thinking of the comment where the `IdDict` contained spurious entries)? Can we just define `Base.haskey(gs::Grads, p) = p ∈ keys(gs)` instead?

Reply:
good catch